### ISSUE TITLE
isDocumentArray does not work for a property of type mongoose.Types.Array<Ref>

### ISSUE THREAD

**Issue by rubenvereecken** [typegoose/typegoose#278]
## Versions

- NodeJS: v12.x
- Typegoose(NPM): @typegoose/typegoose@7.1.3 (git://github.com/typegoose/typegoose.git#2242b63b6ea77500b9b21184a68eb5219bedb9a7)
- mongoose: 5.9.10

## What is the Problem?

`isDocumentArray` does not seem to be working for a property of type `mongoose.Types.Array<Ref<Something>>`.

## Code Example

I tried to write a self-contained code example. It does illustrate the problem, but the error I get isn't _quite_ the same as in my real code base. Here goes.

```typescript
import {WhatIsIt} from '@typegoose/typegoose/lib/internal/constants';
import {DocumentType, isDocumentArray, mongoose, prop, Ref} from '@typegoose/typegoose';

export class Animal {}

function checkAnimal(animal: DocumentType<Animal>) { }

export class User {
	@prop({required: true, ref: Animal, refType: mongoose.Schema.Types.ObjectId}, WhatIsIt.ARRAY)
	animals: mongoose.Types.Array<Ref<Animal>>;

	public checkAnimals(this: DocumentType<User>) {
		const animals: mongoose.Types.Array<Ref<Animal>> = this.animals;
		if (isDocumentArray(this.animals)) {
                        // this here is the issue
			this.animals.map(animal => checkAnimal(animal))
		}
	}
}
```
`tsc` gives me the following error:
```
error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'DocumentType<Animal>'.
  Type '{}' is missing the following properties from type 'Document': increment, model, $isDeleted, remove, and 55 more.

17              this.animals.map(animal => checkAnimal(animal))
                                                       ~~~~~~
```
Now, the core of the error is the same: `animal` should be recognised as DocumentType there I think. I'm not typescript-savvy enough to know why it says the type is `unknown`, I would have expected it to be `Ref<Animal, ObjectId`. But that's not the actual issue here. I can update the example if you know how.

To compare, here's my real code's error. Similar setup.
```
Argument of type 'Ref<User, string>' is not assignable to parameter of type 'DocumentType<User>'.
  Type 'string' is not assignable to type 'DocumentType<User>'.
    Type 'string' is not assignable to type 'Document'.

245             const profiles = this.attendees.map(attendee => UserProfile.fromUsers(attendee, user))
                                                                                      ~~~~~~~~
```

## Do you know *why* it happenes?
I had a look at the source and it's probably because `mongoose.Types.Array<Ref>` isn't unified with `Ref<>[]` or something.

**Comment by hasezoey** [comment]
some things i noticed:
- you made the constant `animals`, but never used it
- when you want to use an *empty class* (like `Animal` in this case), `Ref` will somehow be `unkown`, so add an dummy value, and everything will work again
- typegoose dosnt handle arrays that are not `Ref<>[]` (like `mongoose.Types.Array<Ref<>>`) at the moment

will be fixed in 7.2.0

**Comment by hasezoey** [comment]
its fixed in a660ab66fe3e544dc00c9c300b5e7f511881f1ec, for now you can install typegoose as an git-repository until 7.2.0 releases

**Comment by rubenvereecken** [comment]
Oh wow that's great, thanks! And I also learned today type guards don't need a body. Your continued efforts are seriously much appreciated.
